sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/InvalidCastToInterface.cs

```diff
@@ -132,4 +131,44 @@ public void TestMethod(object obj)
             return null;
         }
     }
+
+    interface IFoo { }
+    interface IBar { }
+
+    class Foo : IFoo { }
+    class Bar : IBar { }
+    class FooBar : IFoo, IBar { }
+    sealed class FinalBar : IBar { }
+
+    class Other
+    {
+        public void Method()

```

Comment:
I've added the test for generic arguments. From what I see, the existing cases already cover generics (or compiler does). Note for now this reports bugs, not code smells.